### ISSUE TITLE
Implement the backend application

### ISSUE THREAD

**Issue by pgesek** [benetech/MathShare#311]
This ticket is about starting the development of the Backend application.

**Acceptance Criteria:**
* Generate a Spring Boot project, working with Java 9 (https://start.spring.io/ can be used)
* Should build with Java 9
* Should have appropriate group and artifact id for this project
* Add the dependencies: Web, Actuator, JPA, Rest Docs, DevTools and any other we need
* Builds with Maven

**Comment by oskarhinc** [comment]
@johnhbenetech should we use another repository for the backend, or just stay in this one?

**Comment by oskarhinc** [comment]
@pgesek as Java 9 has reached end of support, and Oracle recommends switching to Java 10 SE, I think we should do it. Please let me know if you disagree.

**Comment by pgesek** [comment]
@oskarhinc yes, the plan is for Oracle to release Java versions more rapidly. We should use the latest stable one.

**Comment by johnhbenetech** [comment]
@oskarhinc Are there pros/cons to either solution? I imagine if we had a different repo for the backend, then the build process could just pull the necessary front end code from this repo?

I went ahead and created the repo, if we'd rather not use it i can delete/rename it:
https://github.com/benetech/MathShareSpring

**Comment by pgesek** [comment]
@johnhbenetech I think we can keep the backend/frontend independent. In that case the frontend can continue to be deployed to the s3 backend, while the backend will run on its Docker container independently, only exposing the API. I feel like this will make us more flexible for future integrations etc.
So i say we use the separate repo for the backend.

If we would want to bundle it all together into one Boot application, then we should still be able to pull the UI using an npm plugin.

**Comment by oskarhinc** [comment]
@johnhbenetech both solutions have benefits and disadvantages. Out team discussed it today and we think that using different repo will make building the separate applications easier and more intuitive. The name for this repo could be discussed at today's meeting.

**Comment by oskarhinc** [comment]
@johnhbenetech we can discuss the name of this repo at today's meeting. I agree with @pgesek - separate repositories will make building the applications easier and more intuitive.

**Comment by oskarhinc** [comment]
Thanks for the research @johnhbenetech!
